4.9 build fails with --enable-shared=no --enable-static=yes

```
libtool: link: x86_64-pc-linux-gnu-gcc -march=native -mtune=native -O2 -pipe -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -o sulogin sulogin.o  -Wl,--as-needed ../libmisc/.libs/libmisc.a ../lib/.libs/libshadow.a -lcrypt
x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I..  -I../lib -I../libmisc -I../libsubid   -march=native -mtune=native -O2 -pipe -c -o list_subid_ranges-list_subid_ranges.o `test -f 'list_subid_ranges.c' || echo './'`list_subid_ranges.c
/bin/sh ../libtool  --tag=CC   --mode=link x86_64-pc-linux-gnu-gcc  -march=native -mtune=native -O2 -pipe  -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -Wl,--as-needed -o list_subid_ranges list_subid_ranges-list_subid_ranges.o ../lib/libshadow.la ../libmisc/libmisc.la ../libsubid/libsubid.la        -lcrypt  -ldl
libtool:   error: cannot find name of link library for '../libsubid/libsubid.la'
make[2]: *** [Makefile:1085: list_subid_ranges] Error 1
make[2]: Leaving directory '/var/tmp/portage/sys-apps/shadow-4.9/work/shadow-4.9/src'
make[1]: *** [Makefile:452: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/sys-apps/shadow-4.9/work/shadow-4.9'
make: *** [Makefile:380: all] Error 2
```

--- a/configure.ac
+++ b/configure.ac
@@ -321,6 +321,8 @@ if test "$with_sha_crypt" = "yes"; then
 	AC_DEFINE(USE_SHA_CRYPT, 1, [Define to allow the SHA256 and SHA512 password encryption algorithms])
 fi
 
+AM_CONDITIONAL(ENABLE_SHARED, test "x$enable_shared" = "xyes")
+
 AM_CONDITIONAL(USE_BCRYPT, test "x$with_bcrypt" = "xyes")
 if test "$with_bcrypt" = "yes"; then
 	AC_DEFINE(USE_BCRYPT, 1, [Define to allow the bcrypt password encryption algorithm])

--- a/libsubid/Makefile.am
+++ b/libsubid/Makefile.am
@@ -1,6 +1,8 @@
 lib_LTLIBRARIES = libsubid.la
+if ENABLE_SHARED
 libsubid_la_LDFLAGS = -Wl,-soname,libsubid.so.@LIBSUBID_ABI@ \
 	-shared -version-info @LIBSUBID_ABI_MAJOR@
+endif
 libsubid_la_SOURCES = api.c
 
 pkginclude_HEADERS = subid.h

